gnocchi/rest/aggregates/api.py

```diff
@@ -232,8 +233,8 @@ def groupper(r):
 
         else:
             try:
-                metric_ids = [six.text_type(utils.UUID(m))
-                              for (m, a) in references]
+                metric_ids = set([six.text_type(utils.UUID(m))
```

Comment:
You should remove the `[]` to build the set without an intermediate list.

Reply:
done